app/core/queue.py

```diff
@@ -0,0 +1,101 @@
+from typing import Iterable, Any, Protocol
+from queue import SimpleQueue
+
+import sys
+import pika
+
+from .settings import Settings, QueueMode
+
+__all__ = [
+    "IMessageQueue",
+    "MemoryMessageQueue",
+    "RabbitMQMessageQueue",
+    "get_message_queue",
+]
+
+settings = Settings()
+
+
+class IMessageQueue(Protocol):
```

Comment:
the queue interface is by no means complete, it's just something simple to get started.